icicle/primitives/field.cuh

```diff
@@ -75,11 +109,17 @@ public:
     return Field{omega_inv.storages[logn - 1]};
   }
 
-  static HOST_INLINE Field inv_log_size(uint32_t logn)
+  static HOST_DEVICE_INLINE Field inv_log_size(uint32_t logn)
   {
     if (logn == 0) { return Field{CONFIG::one}; }
-
+#ifndef __CUDA_ARCH__
     if (logn > CONFIG::omegas_count) THROW_ICICLE_ERR(IcicleError_t::InvalidArgument, "Field: Invalid inv index");
+#else
+    if (logn > CONFIG::omegas_count)
+      printf(
```

Comment:
error code

Reply:
should return error code or throw instead of printing or exit(1) etc, the code has own error handling - and should be following as implemented in main, there is a bit of  a mix right now

Reply:
the failed assert has printout with code line and failed condition - otherwise it will print just assert(false)